opencv_world: Unable to compile opencv 3.1 with nonfree xfeatures2d module

I'm unable to compile Opencv with xFeatures2D contrib module without CUDA support.

Running the following cmake comand:

```
cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/usr/local -DOPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules -DWITH_EIGEN=ON -DWITH_FFMPEG=ON -DWITH_JASPER=ON -DWITH_JPEG=ON -DWITH_PNG=ON -DWITH_V4L=ON -DWITH_LIBV4L=ON -DWITH_XINE=ON -DBUILD_PACKAGE=ON -DBUILD_OPENCV_PYTHON3=ON -DPYTHON2_NUMPY_INCLUDE_DIRS=/usr/lib/python2.7/dist-packages/numpy/core/include -DPYTHON3_NUMPY_INCLUDE_DIRS=/usr/lib/python3/dist-packages/numpy/core/include -DINSTALL_CREATE_DISTRIB=ON -DWITH_CUDA=OFF ..
```

Gives that:

```
--   OpenCV modules:
--     To be built:                 core flann imgproc ml photo video imgcodecs shape videoio highgui objdetect superres features2d calib3d stitching videostab world xfeatures2d ximgproc xobjdetect xphoto aruco bgsegm bioinspired ccalib dnn dpm face fuzzy line_descriptor optflow plot reg rgbd saliency stereo structured_light surface_matching text ts datasets tracking
--     Disabled:                    contrib_world
--     Disabled by dependency:      -
--     Unavailable:                 cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev java python2 viz cvv hdf matlab sfm
```

When compiling the result are this:

```
In file included from /home/user/deps/opencv/modules/stitching/include/opencv2/stitching.hpp:49:0,
                 from /home/user/deps/opencv/modules/stitching/src/precomp.hpp:59,
                 from /home/user/deps/opencv/modules/stitching/src/matchers.cpp:43:
/home/user/deps/opencv/modules/stitching/include/opencv2/stitching/detail/matchers.hpp:52:42: fatal error: opencv2/xfeatures2d/cuda.hpp: No such file or directory
```

Ok, after removing the stitching module (do i really need to???)...

```
cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/usr/local -DOPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules -DWITH_EIGEN=ON -DWITH_FFMPEG=ON -DWITH_JASPER=ON -DWITH_JPEG=ON -DWITH_PNG=ON -DWITH_V4L=ON -DWITH_LIBV4L=ON -DWITH_XINE=ON -DBUILD_PACKAGE=ON -DBUILD_OPENCV_PYTHON3=ON -DPYTHON2_NUMPY_INCLUDE_DIRS=/usr/lib/python2.7/dist-packages/numpy/core/include -DPYTHON3_NUMPY_INCLUDE_DIRS=/usr/lib/python3/dist-packages/numpy/core/include -DINSTALL_CREATE_DISTRIB=ON -DWITH_CUDA=OFF -DBUILD_opencv_stitching=OFF ..
```

Gives me this error:

```
/home/user/deps/opencv/modules/world/src/precomp.hpp:57:43: fatal error: opencv2/xfeatures2d/nonfree.hpp: No such file or directory
```

Which IS present as seems below:

```
user@pc:~/deps/opencv/release$ ls ../../opencv_contrib/modules/xfeatures2d/include/opencv2/
xfeatures2d  xfeatures2d.hpp
```

So, what should i do next? And why do stitching modules crashes even with CUDA explicity disabled? :)

Tried the proposed solution (below) and almost worked...

```
cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/usr/local -DOPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules -DWITH_EIGEN=ON -DWITH_FFMPEG=ON -DWITH_JASPER=ON -DWITH_JPEG=ON -DWITH_PNG=ON -DWITH_V4L=ON -DWITH_LIBV4L=ON -DWITH_XINE=ON -DBUILD_PACKAGE=ON -DBUILD_OPENCV_PYTHON3=ON -DPYTHON2_NUMPY_INCLUDE_DIRS=/usr/lib/python2.7/dist-packages/numpy/core/include -DPYTHON3_NUMPY_INCLUDE_DIRS=/usr/lib/python3/dist-packages/numpy/core/include -DBUILD_opencv_world=OFF -DWITH_CUDA=OFF .. 
```

At "100%" the following error shows up:
```
Linking CXX executable ../../bin/opencv_interactive-calibration
/usr/bin/ld: CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o: undefined reference to symbol '_ZN2cv12cornerSubPixERKNS_11_InputArrayERKNS_17_InputOutputArrayENS_5Size_IiEES7_NS_12TermCriteriaE'
/home/user/deps/opencv/release/lib/libopencv_imgproc.so.3.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/build.make:219: recipe for target 'bin/opencv_interactive-calibration' failed
make[2]: *** [bin/opencv_interactive-calibration] Error 1
CMakeFiles/Makefile2:6722: recipe for target 'apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all' failed
make[1]: *** [apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all] Error 2
Makefile:147: recipe for target 'all' failed
make: *** [all] Error 2
```

My cmake ...  outputs:
```
-- Detected version of GNU GCC: 49 (409)
-- FP16: Feature disabled
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found suitable version "1.2.8", minimum required is "1.2.3") 
-- Could NOT find TIFF (missing:  TIFF_LIBRARY TIFF_INCLUDE_DIR) 
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
-- checking for module 'gtk+-3.0'
--   package 'gtk+-3.0' not found
-- checking for module 'gstreamer-video-1.0'
--   package 'gstreamer-video-1.0' not found
-- checking for module 'gstreamer-app-1.0'
--   package 'gstreamer-app-1.0' not found
-- checking for module 'gstreamer-riff-1.0'
--   package 'gstreamer-riff-1.0' not found
-- checking for module 'gstreamer-pbutils-1.0'
--   package 'gstreamer-pbutils-1.0' not found
-- checking for module 'gstreamer-base-0.10'
--   package 'gstreamer-base-0.10' not found
-- checking for module 'gstreamer-video-0.10'
--   package 'gstreamer-video-0.10' not found
-- checking for module 'gstreamer-app-0.10'
--   package 'gstreamer-app-0.10' not found
-- checking for module 'gstreamer-riff-0.10'
--   package 'gstreamer-riff-0.10' not found
-- checking for module 'gstreamer-pbutils-0.10'
--   package 'gstreamer-pbutils-0.10' not found
-- Looking for linux/videodev.h
-- Looking for linux/videodev.h - not found
-- Looking for linux/videodev2.h
-- Looking for linux/videodev2.h - found
-- Looking for sys/videoio.h
-- Looking for sys/videoio.h - not found
-- Looking for libavformat/avformat.h
-- Looking for libavformat/avformat.h - found
-- Looking for ffmpeg/avformat.h
-- Looking for ffmpeg/avformat.h - not found
-- checking for module 'libgphoto2'
--   package 'libgphoto2' not found
-- A library with BLAS API found.
-- A library with LAPACK API found.
-- found IPP (ICV version): 9.0.1 [9.0.1]
-- at: /home/user/deps/opencv/3rdparty/ippicv/unpack/ippicv_lnx
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
-- To enable PlantUML support, set PLANTUML_JAR environment variable or pass -DPLANTUML_JAR=<filepath> option to cmake
-- Found apache ant 1.9.4: /usr/bin/ant
-- Could NOT find JNI (missing:  JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH) 
-- Could NOT find Matlab (missing:  MATLAB_MEX_SCRIPT MATLAB_INCLUDE_DIRS MATLAB_ROOT_DIR MATLAB_LIBRARIES MATLAB_LIBRARY_DIRS MATLAB_MEXEXT MATLAB_ARCH MATLAB_BIN) 
-- VTK is not found. Please set -DVTK_DIR in CMake to VTK build directory, or to VTK install subdirectory with VTKConfig.cmake file
-- Caffe:   NO
-- Protobuf:   YES
-- Glog:   NO
-- Could NOT find HDF5 (missing:  HDF5_LIBRARIES HDF5_INCLUDE_DIRS) 
-- Module opencv_sfm disabled because the following dependencies are not found: Glog/Gflags
-- The protocol buffer compiler and libprotobuf were found
-- Tesseract:   NO
-- 
-- General configuration for OpenCV 3.1.0-dev =====================================
--   Version control:               3.1.0-1471-gdd379ec
-- 
--   Extra modules:
--     Location (extra):            /home/user/deps/opencv_contrib/modules
--     Version control (extra):     3.1.0-386-gc1101a0
-- 
--   Platform:
--     Timestamp:                   2016-10-10T18:05:54Z
--     Host:                        Linux 3.16.0-4-amd64 x86_64
--     CMake:                       3.0.2
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               RELEASE
-- 
--   C/C++:
--     Built as dynamic libs?:      YES
--     C++ Compiler:                /usr/bin/c++  (ver 4.9.2)
--     C++ flags (Release):         -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C++ flags (Debug):           -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /usr/bin/cc
--     C flags (Release):           -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wno-narrowing -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -O3 -DNDEBUG  -DNDEBUG
--     C flags (Debug):             -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wno-narrowing -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):      
--     Linker flags (Debug):        
--     Precompiled headers:         NO
--     Extra dependencies:          dl m pthread rt
--     3rdparty dependencies:
-- 
--   OpenCV modules:
--     To be built:                 core flann imgproc ml photo reg surface_matching video dnn fuzzy imgcodecs shape videoio highgui objdetect plot superres ts xobjdetect xphoto bgsegm bioinspired dpm face features2d line_descriptor saliency text calib3d ccalib datasets rgbd stereo structured_light tracking videostab xfeatures2d ximgproc aruco optflow python2 python3
--     Disabled:                    stitching world contrib_world
--     Disabled by dependency:      -
--     Unavailable:                 cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev java viz cvv hdf matlab sfm
-- 
--   GUI: 
--     QT:                          NO
--     GTK+ 2.x:                    YES (ver 2.24.25)
--     GThread :                    YES (ver 2.42.1)
--     GtkGlExt:                    NO
--     OpenGL support:              NO
--     VTK support:                 NO
-- 
--   Media I/O: 
--     ZLib:                        /usr/lib/x86_64-linux-gnu/libz.so (ver 1.2.8)
--     JPEG:                        /usr/lib/x86_64-linux-gnu/libjpeg.so (ver )
--     WEBP:                        build (ver 0.3.1)
--     PNG:                         /usr/lib/x86_64-linux-gnu/libpng.so (ver 1.2.50)
--     TIFF:                        build (ver 42 - 4.0.2)
--     JPEG 2000:                   /usr/lib/x86_64-linux-gnu/libjasper.so (ver 1.900.1)
--     OpenEXR:                     build (ver 1.7.1)
--     GDAL:                        NO
--     GDCM:                        NO
-- 
--   Video I/O:
--     DC1394 1.x:                  NO
--     DC1394 2.x:                  YES (ver 2.2.3)
--     FFMPEG:                      YES
--       codec:                     YES (ver 56.60.100)
--       format:                    YES (ver 56.40.101)
--       util:                      YES (ver 54.31.100)
--       swscale:                   YES (ver 3.1.101)
--       resample:                  YES (ver 2.1.0)
--       gentoo-style:              YES
--     GStreamer:                   NO
--     OpenNI:                      NO
--     OpenNI PrimeSensor Modules:  NO
--     OpenNI2:                     NO
--     PvAPI:                       NO
--     GigEVisionSDK:               NO
--     UniCap:                      NO
--     UniCap ucil:                 NO
--     V4L/V4L2:                    Using libv4l1 (ver 1.10.0) / libv4l2 (ver 1.10.0)
--     XIMEA:                       NO
--     Xine:                        YES (ver 1.2.6)
--     gPhoto2:                     NO
-- 
--   Parallel framework:            pthreads
-- 
--   Other third-party libraries:
--     Use IPP:                     9.0.1 [9.0.1]
--          at:                     /home/user/deps/opencv/3rdparty/ippicv/unpack/ippicv_lnx
--     Use IPP Async:               NO
--     Use VA:                      NO
--     Use Intel VA-API/OpenCL:     NO
--     Use Lapack:                  NO
--     Use Eigen:                   YES (ver 3.2.2)
--     Use Cuda:                    NO
--     Use OpenCL:                  YES
--     Use OpenVX:                  NO
--     Use custom HAL:              NO
-- 
--   OpenCL:                        <Dynamic loading of OpenCL library>
--     Include path:                /home/user/deps/opencv/3rdparty/include/opencl/1.2
--     Use AMDFFT:                  NO
--     Use AMDBLAS:                 NO
-- 
--   Python 2:
--     Interpreter:                 /usr/bin/python2.7 (ver 2.7.9)
--     Libraries:                   /usr/lib/x86_64-linux-gnu/libpython2.7.so (ver 2.7.9)
--     numpy:                       /usr/lib/python2.7/dist-packages/numpy/core/include (ver 1.8.2)
--     packages path:               lib/python2.7/dist-packages
-- 
--   Python 3:
--     Interpreter:                 /usr/bin/python3.4 (ver 3.4.2)
--     Libraries:                   /usr/lib/x86_64-linux-gnu/libpython3.4m.so (ver 3.4.2)
--     numpy:                       /usr/lib/python3/dist-packages/numpy/core/include (ver 1.8.2)
--     packages path:               lib/python3.4/dist-packages
-- 
--   Python (for build):            /usr/bin/python2.7
-- 
--   Java:
--     ant:                         /usr/bin/ant (ver 1.9.4)
--     JNI:                         NO
--     Java wrappers:               NO
--     Java tests:                  NO
-- 
--   Matlab:                        Matlab not found or implicitly disabled
-- 
--   Documentation:
--     Doxygen:                     NO
--     PlantUML:                    NO
-- 
--   Tests and samples:
--     Tests:                       YES
--     Performance tests:           YES
--     C/C++ Examples:              NO
-- 
--   Install path:                  /usr/local
-- 
--   cvconfig.h is in:              /home/user/deps/opencv/release
-- -----------------------------------------------------------------
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/deps/opencv/release
```

The stderr returned:
```
/home/user/deps/opencv/modules/imgproc/src/contours.cpp: In function ‘CvSeq* cvFindNextContour(CvContourScanner)’:
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:849:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:667:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:574:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:930:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:745:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/usr/bin/ld: CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o: undefined reference to symbol '_ZN2cv12cornerSubPixERKNS_11_InputArrayERKNS_17_InputOutputArrayENS_5Size_IiEES7_NS_12TermCriteriaE'
/home/user/deps/opencv/release/lib/libopencv_imgproc.so.3.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [bin/opencv_interactive-calibration] Error 1
make[1]: *** [apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all] Error 2
make: *** [all] Error 2
```

The stdout is attached: [cv_stdout.txt](https://github.com/opencv/opencv/files/520556/cv_stdout.txt)

Returned stderr:
```
/home/user/deps/opencv/modules/imgproc/src/contours.cpp: In function ‘CvSeq* cvFindNextContour(CvContourScanner)’:
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:849:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:667:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:574:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:930:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/home/user/deps/opencv/modules/imgproc/src/contours.cpp:745:37: warning: array subscript is above array bounds [-Warray-bounds]
                 i4 = i3 + deltas[++s];
                                     ^
/usr/bin/ld: CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o: undefined reference to symbol '_ZN2cv12cornerSubPixERKNS_11_InputArrayERKNS_17_InputOutputArrayENS_5Size_IiEES7_NS_12TermCriteriaE'
/home/user/deps/opencv/release/lib/libopencv_imgproc.so.3.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [bin/opencv_interactive-calibration] Error 1
make[1]: *** [apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all] Error 2
make: *** [all] Error 2
```

stdout (only last part... everything else seemed ok):
```
Linking CXX executable ../../bin/opencv_interactive-calibration
cd /home/user/deps/opencv/release/apps/interactive-calibration && /usr/bin/cmake -E cmake_link_script CMakeFiles/opencv_interactive-calibration.dir/link.txt --verbose=1
/usr/bin/c++      -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG       CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o CMakeFiles/opencv_interactive-calibration.dir/calibPipeline.cpp.o CMakeFiles/opencv_interactive-calibration.dir/calibController.cpp.o CMakeFiles/opencv_interactive-calibration.dir/rotationConverters.cpp.o CMakeFiles/opencv_interactive-calibration.dir/parametersController.cpp.o CMakeFiles/opencv_interactive-calibration.dir/main.cpp.o  -o ../../bin/opencv_interactive-calibration -rdynamic -ldl -lm -lpthread -lrt ../../../3rdparty/ippicv/unpack/ippicv_lnx/lib/intel64/libippicv.a ../../lib/libopencv_core.so.3.1.0 ../../lib/libopencv_aruco.so.3.1.0 ../../lib/libopencv_highgui.so.3.1.0 ../../lib/libopencv_calib3d.so.3.1.0 ../../lib/libopencv_videoio.so.3.1.0 ../../lib/libopencv_highgui.so.3.1.0 ../../lib/libopencv_videoio.so.3.1.0 ../../lib/libopencv_core.so.3.1.0 -Wl,-rpath,/home/user/deps/opencv/release/lib: -Wl,-rpath-link,/home/user/deps/opencv/release/lib 
apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/build.make:219: recipe for target 'bin/opencv_interactive-calibration' failed
make[2]: Leaving directory '/home/user/deps/opencv/release'
CMakeFiles/Makefile2:6722: recipe for target 'apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all' failed
make[1]: Leaving directory '/home/user/deps/opencv/release'
Makefile:147: recipe for target 'all' failed
```

Still with the same problem...

``` 
Linking CXX executable ../../bin/opencv_interactive-calibration
cd /home/user/deps/opencv/release/apps/interactive-calibration && /usr/bin/cmake -E cmake_link_script CMakeFiles/opencv_interactive-calibration.dir/link.txt --verbose=1
/usr/bin/c++      -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -mno-avx -msse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -ffunction-sections -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG       CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o CMakeFiles/opencv_interactive-calibration.dir/calibPipeline.cpp.o CMakeFiles/opencv_interactive-calibration.dir/calibController.cpp.o CMakeFiles/opencv_interactive-calibration.dir/rotationConverters.cpp.o CMakeFiles/opencv_interactive-calibration.dir/parametersController.cpp.o CMakeFiles/opencv_interactive-calibration.dir/main.cpp.o  -o ../../bin/opencv_interactive-calibration -rdynamic -ldl -lm -lpthread -lrt ../../../3rdparty/ippicv/unpack/ippicv_lnx/lib/intel64/libippicv.a ../../lib/libopencv_core.so.3.1.0 ../../lib/libopencv_imgproc.so.3.1.0 ../../lib/libopencv_aruco.so.3.1.0 ../../lib/libopencv_highgui.so.3.1.0 ../../lib/libopencv_calib3d.so.3.1.0 ../../lib/libopencv_videoio.so.3.1.0 ../../lib/libopencv_highgui.so.3.1.0 ../../lib/libopencv_videoio.so.3.1.0 ../../lib/libopencv_imgproc.so.3.1.0 ../../lib/libopencv_core.so.3.1.0 -Wl,-rpath,/home/user/deps/opencv/release/lib: -Wl,-rpath-link,/home/user/deps/opencv/release/lib 
/usr/bin/ld: CMakeFiles/opencv_interactive-calibration.dir/frameProcessor.cpp.o: undefined reference to symbol '_ZN2cv18SimpleBlobDetector6createERKNS0_6ParamsE'
/home/user/deps/opencv/release/lib/libopencv_features2d.so.3.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/build.make:221: recipe for target 'bin/opencv_interactive-calibration' failed
make[2]: *** [bin/opencv_interactive-calibration] Error 1
make[2]: Leaving directory '/home/user/deps/opencv/release'
CMakeFiles/Makefile2:6723: recipe for target 'apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all' failed
make[1]: *** [apps/interactive-calibration/CMakeFiles/opencv_interactive-calibration.dir/all] Error 2
make[1]: Leaving directory '/home/user/deps/opencv/release'
Makefile:147: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/apps/interactive-calibration/CMakeLists.txt
+++ b/apps/interactive-calibration/CMakeLists.txt
@@ -1,4 +1,4 @@
-set(OPENCV_INTERACTIVECALIBRATION_DEPS opencv_core opencv_aruco opencv_highgui opencv_calib3d opencv_videoio)
+set(OPENCV_INTERACTIVECALIBRATION_DEPS opencv_core opencv_imgproc opencv_features2d opencv_aruco opencv_highgui opencv_calib3d opencv_videoio)
 ocv_check_dependencies(${OPENCV_INTERACTIVECALIBRATION_DEPS})
 
 if(NOT OCV_DEPENDENCIES_FOUND)

